src/main/java/io/confluent/idesidecar/restapi/proxy/FlinkDataPlaneAuthenticationProcessor.java

```diff
@@ -0,0 +1,63 @@
+package io.confluent.idesidecar.restapi.proxy;
+
+import static io.vertx.core.http.HttpHeaders.AUTHORIZATION;
+
+import io.confluent.idesidecar.restapi.connections.CCloudConnectionState;
+import io.confluent.idesidecar.restapi.exceptions.ProcessorFailedException;
+import io.confluent.idesidecar.restapi.processors.Processor;
+import io.vertx.core.Future;
+import io.vertx.core.MultiMap;
+import jakarta.enterprise.context.ApplicationScoped;
+
+/**
+ * Processor for handling data plane requests to services like Flink SQL and Catalog.
+ */
+@ApplicationScoped
+public class FlinkDataPlaneAuthenticationProcessor extends Processor<ProxyContext, Future<ProxyContext>> {
+
+  @Override
+  public Future<ProxyContext> process(ProxyContext context) {
+    var connectionState = context.getConnectionState();
+
+    if (!(connectionState instanceof CCloudConnectionState cCloudConnection)) {
+      return Future.failedFuture(
+          new ProcessorFailedException(
+              context.fail(400, "Bad Request: Non-CCloud connections are not supported")));
+    }
+
+    // Check if this is a data plane request
+    if (!isDataPlaneRequest(context)) {
+      // Skip non-data plane requests
+      return next().process(context);
+    }
```

Comment:
I think we can leave this check out, so that we allow users to configure data plane routes via the application config.